### PR TITLE
chore: Clean up unused and completed feature flags

### PR DESCRIPTION
**Pull request by hetunandu**
## Description
Clean up unused feature flags

- LINTING
- APP_TEMPLATE
- JS_EDITOR 
- MULTIPLAYER
- SNIPPET
- TEMPLATES_PHASE_2
- RBAC
- CONTEXT_SWITCHING
- USAGE_AND_BILLING
- DATASOURCE_ENVIRONMENTS

#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)

## Testing
- [ ] Manual
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Test-plan-implementation#speedbreaker-features-to-consider-for-every-change) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans/_edit#areas-of-interest)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/server/appsmith-server/src/main/resources/features/init-flags.yml [hunk 1]
```diff
@@ -8,68 +8,6 @@ ff4j:
...
-    - uid: CONTEXT_SWITCHING
-      enable: true
-      description: Restoring old context while navigating across the app
-      flipstrategy:
-        class: org.ff4j.strategy.PonderationStrategy
-        param:
-          - name: weight
-            value: 1
-
-    - uid: DATASOURCE_ENVIRONMENTS
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/resources/features/init-flags.yml:65]: We do need this flag please :)

**Review by nidhi-nair — DISMISSED** [review]

File: app/client/src/components/formControls/utils.test.ts [hunk 2]
```diff
@@ -289,13 +289,13 @@ describe("caculateIsHidden test", () => {
...
       comparison: "EQUALS",
       value: "Name",
-      flagValue: "APP_TEMPLATE",
+      flagValue: "TEST_FLAG",
     };
     const hiddenFalsy: HiddenType = {
       path: "name",
       comparison: "EQUALS",
       value: "Different Name",
-      flagValue: "APP_TEMPLATE",
```
Comment (akash-codemonk) [inline app/client/src/components/formControls/utils.test.ts:298]: Is this field related to feature flag?

**Review by akash-codemonk — APPROVED** [review]
